android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsFullScreenAdModule.kt

```diff
@@ -149,6 +151,27 @@ abstract class ReactNativeGoogleMobileAdsFullScreenAdModule<T>(
         var eventType = ReactNativeGoogleMobileAdsEvent.GOOGLE_MOBILE_ADS_EVENT_LOADED
         var data: WritableMap? = null
 
+        var paidEventListener = OnPaidEventListener { adValue ->
+          val payload = Arguments.createMap()
+          payload.putDouble("value", 1e-6 * adValue.getValueMicros());
+          payload.putDouble("precision", 1.0 * adValue.getPrecisionType());
```

Comment:
interesting difference between java and kotlin that you have to do the `1.0 * ...` thing here

Reply:
Somehow, Java is fine to auto-cast integer into double, and Kotlin had issues with it (perhaps because, if I got it right, it considers it not just as a integer, but as a sort of an enumerator value). Also, I don't really know Kotlin syntax, I believe I tried something like `(double)...`, `Double(...)`, _etc._, it did not work, and `1.0 * ...` was the first working way to cast it into double I found.

Reply:
I'm still learning Kotlin myself and not very good at it, so I was just mentioning it. Very subtle difference!